cmd/object-handlers.go

```diff
@@ -1197,15 +1194,6 @@ func (api objectAPIHandlers) PutObjectHandler(w http.ResponseWriter, r *http.Req
 		}
 	}
 
-	if hashError != nil {
```

Comment:
Our current unit tests works such a way that the `putObject()` errors are treated with higher priority. Since unit tests wont run on compression mode we cannot detect such behaviour. 
So by recording the `hashError` and then erroring out if `putObject` err is nil gives higher priority to the putObject errors.
Fine if this is not a problem here.

Reply:
The code was wrong in the first place, you cannot set a value on a variable inside a go-routine and read it outside. You are actively introducing a race, the way to communicate over Pipe is by sending  CloseWithError() this way the error gets propagated through the Pipe to the reader. 

Reader fails appropriately. if you look at the code it was sending error responses after setting some successful encryption headers etc.

Reply:
ok